File: Second Year/Second Semester/Web Programming/ExamPreparation/MockExams/2022-main-day1/JSP-Exam/src/main/java/web/exam/jspexam/model/Template.java
package web.exam.jspexam.model;

public class Template {
    private Integer id;
    private String name;
    private String textContent;

    public Template(Integer id, String name, String textContent) {
        this.id = id;
        this.name = name;
        this.textContent = textContent;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getTextContent() {
        return textContent;
    }

    public void setTextContent(String textContent) {
        this.textContent = textContent;
    }
}

File: Second Year/Second Semester/Web Programming/ExamPreparation/MockExams/2022-main-day1/PHP-Exam/server/model/template.php
<?php
class Template implements JsonSerializable {
    private $id;
    private $name;
    private $textContent;

    public function __construct(int $id, string $name, string $textContent) {
        $this->id = $id;
        $this->name = $name;
        $this->textContent = $textContent;
    }

    public function getId(): int {
        return $this->id;
    }
 
    public function getName() : string {
        return $this->name;
    }

    public function getTextContent() : string {
        return $this->textContent;
    }

    public function jsonSerialize() {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'textContent' => $this->textContent,
        ];
    }
}
?>
